### PR TITLE
Package: Johnny-five - Added objects MotorPins and Motors

### PR DESCRIPTION
**Pull request by xtrimsystems**
Added value object MotorPins for pins in MotorOption, added class Motors and update SensorOption to use new attribute enabled

Please fill in this template.

- [x ] Use a meaningful title for the pull request. Include the name of the package modified.
- [x ] Test the change in your own code. (Compile and run.)
- [x ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x ] Provide a URL to documentation or source code which provides context for the suggested changes: http://johnny-five.io/api/motors/#api
- [ ] Increase the version number in the header if appropriate.
- [n/a ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by xtrimsystems** [comment]
@nakakura @ujvzolee @Workshop2

File: types/johnny-five/index.d.ts [hunk 1]
```diff
@@ -540,6 +548,22 @@ export declare class Motor {
...
+    constructor(option: Array<number> | MotorOption[]);
+
+    readonly isOn: boolean;
+
+    forward(speed: number): void;
+    fwd(speed: number): void;
+    reverse(speed: number): void;
+    rev(speed: number): void;
+    start(): void;
+    start(speed: number): void;
```
Comment (DanielRosenwasser) [inline types/johnny-five/index.d.ts]: Make this an optional argument and get rid of the above overload.

File: types/johnny-five/index.d.ts [hunk 2]
```diff
@@ -540,6 +548,22 @@ export declare class Motor {
     release(): void;
 }
 
+export declare class Motors {
+    constructor(option: Array<number> | MotorOption[]);
```
Comment (DanielRosenwasser) [inline types/johnny-five/index.d.ts]: Nit: `number[] | MotorOption[]`

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: types/johnny-five/index.d.ts (hunk 1 above)
Comment (xtrimsystems) [reply types/johnny-five/index.d.ts]: Ok, I just copied from the Motor class, so I will also make it optional and delete the overload there

**Comment by DanielRosenwasser** [comment]
Thanks!

**Pull request merged**